src/main/java/seedu/address/logic/commands/IncrementCommand.java

```diff
@@ -0,0 +1,117 @@
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Department;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Id;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Role;
+import seedu.address.model.person.Salary;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Increments the salary of all persons in the displayed person list by the given increment.
+ */
+public class IncrementCommand extends Command {
+    public static final String COMMAND_WORD = "increment";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Increases the salary of all persons in the displayed person list by the input value.\n"
+            + "Parameters: INCREMENT\n"
+            + "Example: " + COMMAND_WORD + " 100";
+
+    public static final String MESSAGE_INCREMENT_SUCCESS = "Incremented salary of %1$s person(s) by: %2$s";
+
+    public static final String MESSAGE_INVALID_INCREMENT = "Increment of %1$s causes salary of {%2$s} to fall below 0";
+
+    private final Increment increment;
+
+    /**
+     * Constructs a {@code IncrementCommand}
+     *
+     * @param increment A {@code Increment}.
+     */
+    public IncrementCommand(Increment increment) {
+        this.increment = increment;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+        checkValidIncrement(lastShownList);
+
+        for (Person personToEdit : lastShownList) {
+            Person editedPerson = createdEditedPerson(personToEdit);
+            model.setPerson(personToEdit, editedPerson);
+        }
+
+        return new CommandResult(String.format(MESSAGE_INCREMENT_SUCCESS, lastShownList.size(), increment.toString()));
+    }
+
+    /**
+     * Checks if the persons in the given {@code personList} all have a salary that is greater than or equal to
+     * the {@code increment}.
+     * @throws CommandException If at least one of the persons have a salary that is less than the {@code increment}.
+     */
+    private void checkValidIncrement(List<Person> personList) throws CommandException {
+        requireNonNull(personList);
+        for (Person person : personList) {
+            if (person.getSalary().isNegativeAfterIncrement(increment)) {
+                throw new CommandException(
+                        String.format(MESSAGE_INVALID_INCREMENT, increment, Messages.format(person)));
+            }
+        }
+    }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit} and with the incremented salary.
+     */
+    private Person createdEditedPerson(Person personToEdit) {
```

Comment:
Is the name too similar to the createEditedPerson in EditCommand? Could we change it so that it sounds more distinct?
etc incrementPersonSalary